packages/ilib-po/src/Parser.ts

```diff
@@ -162,15 +178,38 @@ class Parser {
             original: string | undefined,
             sourcePlurals: Plural | undefined,
             translationPlurals: Plural | undefined,
-            category : PluralCategory | undefined;
+            category : PluralCategory | undefined,
+            arrays: ArrayResources | undefined = {},
+            key: string | undefined,
+            type: string | undefined,
+            arrayIndex: number = 0,
+            defaultDataType: string | undefined = this.datatype,
+            datatype: string | undefined,
+            defaultProject: string | undefined = this.projectName,
+            project: string | undefined;
```

Comment:
Can't do that -- these ones have a purpose that is different than the instance-level values. The default data type and project name can be defined in the headers of the po file and can be different than the ones specified parameters to the POFile constructor. We still need a place to store that. The default data type and project are parsed out of the header of the po file on lines 204-211. If they are not there in the file, they fall back to the values for the instance. Maybe I should call them `defaultFileDataType` and `defaultFileProject` instead?

Reply:
Oh, I missed that. So basically, the value attached to the produced resource instance fallback should be 1. from unit 2. from file header 3. from parser instance right? In that case, maybe use the word _file_ instead _default_ (rather than both)